cpp/benchmarks/flow_simulation_ode_seir.cpp

```diff
@@ -32,61 +32,92 @@ namespace mio
 namespace benchmark
 {
 
-using FlowModel = ::mio::oseir::Model;
+using FlowModel = ::mio::oseir::Model<ScalarType>;
 
 using namespace oseir;
 
 // For comparison benchmarks, an old model version that does not provide computation of flows has been reimplemented here.
 // For more details see the original implementation in:
 // https://github.com/SciCompMod/memilio/blob/13555a6b23177d2d4633c393903461a27ce5762b/cpp/models/ode_seir/model.h
```

Comment:
This description is now entirely outdated. I think it should now state that it is a rewrite of the current model adapted to be an compartmental model?

Does this comparison benchmark even make sense after all these changes?

Reply:
I have adapted the documentation accordingly. Why should the benchmark no longer make sense?

Reply:
I mean specifically the benchmark using this FlowlessModel. It was meant to compare the old and new version of the model, but now it compares the "newer" and the "also newer, but calculating flows manually" version. This seems a bit odd to me.

Reply:
You are definitely right that the actual purpose of this benchmarking is no longer given. However, I think the comparison between the use of the `get_flows `and `get_derivatives `function is still interesting. Since the benchmark already exists here anyway, I would be in favor of keeping it.

Reply:
Good point.

I tried the current benchmarks, and confusingly, the new oseir model is both slower (using simulate) and faster (using simulate_flows) than the model with get_derivatives (first benchmark):
```
mio::Simulation on oseir::Model (pre 511 branch) without flows     362732 ns       362672 ns         1927
mio::Simulation on oseir::Model with flows                         472133 ns       472027 ns         1367
mio::FlowSimulation on oseir::Model with flows                     201374 ns       201340 ns         3471
```
I think we might need to look at the FlowModel::get_derivatives again...

Resolving this conversation for now.